### PR TITLE
Fixed anchor categories showing products of disabled subcategories #9002

### PR DESCRIPTION
**Pull request by drashmk**
### Description
Incomplete query for the temporary table `temp_catalog_category_tree_index_xxxx` was causing products  of disabled subcategories to be visible on anchor categories. It was missing the check for `is_active` category attribute on the child category when creating the temporary category. 

### Fixed Issues (if relevant)

#9002: Anchor categories are showing products of disabled subcategories

### Manual testing scenarios
1. Create a simple product 'Test' that is in stock and visible on catalog
2. Create a category A (don't add products to it) 
3. Create a subcategory B (parent is A) and add our 'Test' product to it
4. Goto category A on frontend
5. 'Test' product  is not visible in category A

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/Catalog/Model/Indexer/Category/Product/AbstractAction.php [hunk 1]
```diff
@@ -521,10 +521,11 @@ protected function getTemporaryTreeIndexTableName()
      *
      * Returns the name of the temporary table to use in queries.
      *
+     * @param \Magento\Store\Model\Store $store
      * @return string
      * @since 101.0.0
      */
-    protected function makeTempCategoryTreeIndex()
+    protected function makeTempCategoryTreeIndex(\Magento\Store\Model\Store $store)
```
Comment (sidolov) [inline app/code/Magento/Catalog/Model/Indexer/Category/Product/AbstractAction.php:528]: We can't add/remove any parameters to public and protected method, please, use guide for [backward compatible developpmenet](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/)

File: app/code/Magento/Catalog/Model/Indexer/Category/Product/AbstractAction.php [hunk 2]
```diff
@@ -553,19 +554,27 @@ protected function makeTempCategoryTreeIndex()
...
 
     /**
      * Populate the temporary category tree index table
      *
+     * @param \Magento\Store\Model\Store $store
      * @param string $temporaryName
      * @since 101.0.0
      */
-    protected function fillTempCategoryTreeIndex($temporaryName)
+    protected function fillTempCategoryTreeIndex($temporaryName, \Magento\Store\Model\Store $store)
```
Comment (sidolov) [inline app/code/Magento/Catalog/Model/Indexer/Category/Product/AbstractAction.php:569]: See previous comment

**Review by sidolov — CHANGES_REQUESTED** [review]
Please, cover provided changes at least by unit test

**Comment by sidolov** [comment]
@drashmk , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for collaboration

**Comment by magento-engcom-team** [comment]
Hi @drashmk. Thank you for your contribution. 
Changes from your Pull Request will be available with the upcoming 2.3.0 release.

**Pull request closed without merging**